msprime/trees.py

```diff
@@ -289,11 +289,17 @@ def __init__(
         self._tree = tree
         self._x_coords = {}
         self._y_coords = {}
+        try:
+            self._colour = {node:col for node,col in node_colours.keys()}
```

Comment:
It's simplest and most consistent if we use ``self._branch_colours`` as the instance variable here. Then we just say
```
self._branch_colours = {}
if branch_colours is not None:
    self._branch_colours = node_colours
```